packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -100,7 +100,27 @@ export class SingleThreadedWallet
   readonly walletConfig: ServerWalletConfig;
 
   public static create(walletConfig: IncomingServerWalletConfig): SingleThreadedWallet {
-    return new SingleThreadedWallet(walletConfig);
+    const wallet = new SingleThreadedWallet(walletConfig);
+    // This is an async method so it could continue executing after this method returns
```

Comment:
It might make sense to change `create` to an async method so we avoid this.

However, I was reluctant to change the external wallet API as we'd need to update lots of code in this repo and in `the-graph`